### DISCUSSION TITLE
action buttons and modals

### DISCUSSION THREAD

**Discussion by thowt** [rappasoft/laravel-livewire-tables#576] [Q&A]
Hello,

im trying to create simple table with actions column, then add somewhere at top 1 "Add Record" button and use modals.
So I used format on Column
`Column::make('Actions')
                ->format(function($value, $column, $row) {
                    return view('livewire.pdns.records-table-actions');
                }),`

and defined blade as:
`<a href="#" wire:click.prevent="manageRecords()" class="text-primary-600 font-medium hover:text-primary-900">Manage</a>`

pretty simple, but how to pass rowid to blade? For "manageRecords($id)"

And another question is, where and how to put modal and how to link it "correct" livewire component. 
I mean, is it better to put all this functionality together into one component - table? Or do you recommend to separate it?

Sorry for stupid questions, im new to livewire so i am wondering which setup is better. 
Currently, this wire:click.prevent="manageRecords()" is looking for function in Table component. Is there an option to manually point him to different component - modal?

Thank you for help.

**Comment by rizkhal** [comment]
You can use `wire:click="$emit('event-name', 'some-params-here')"` and listen this event on your other component, or if you want to spesific target component you can use `$emitTo('components-here', 'event-name-here', 'some-params-here')`..

For reusable modal component, you can check [this package](https://github.com/wire-elements/modal)
